src/ray/raylet/dependency_manager.h

```diff
@@ -316,6 +320,9 @@ class DependencyManager : public TaskDependencyManagerInterface {
   /// total will be less than or equal to the size of queued_task_requests_.
   CounterMap<TaskMetricsKey> waiting_tasks_counter_;
 
+  /// Metrics
+  ray::observability::MetricInterface &metric_tasks_;
```

Comment:
I feel like the `MetricInterface` should encode the required tags, otherwise the class definition is basically not self-contained. Not sure the best way to structure this.